.azure-pipelines/client.yml

```diff
@@ -2,54 +2,29 @@ trigger:
 - master
 
 jobs:
-- job: 'Publish'
+- job: 'Build'
 
   pool:
     vmImage: 'ubuntu-16.04'
 
   steps:
   - task: Maven@3
-    displayName: 'Checkstyle'
+    displayName: 'Build and Package'
     inputs:
-      mavenPomFile: 'pom.client.build.xml'
-      goals: 'checkstyle:check'
-      options: '--batch-mode'
-      mavenOptions: '-Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
-      publishJUnitResults: false
-  - task: Maven@3
-    inputs:
-      mavenPomFile: 'pom.client.build.xml'
-      options: '--batch-mode'  #hides dependencies download progress from CI output
-      mavenOptions: '-Xmx3072m -Dmaven.test.skip=true -Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+      mavenPomFile: $(pomFile)
+      goals: 'checkstyle:check javadoc:jar source:jar package'
+      options: '--batch-mode -Dmaven.test.skip=true'
```

Comment:
I dug into the mavenOptions vs Options and mavenOptions ends up being a backdoor to setting an environment variable for maven optimizations https://github.com/Microsoft/azure-pipelines-tasks/blob/master/Tasks/MavenV3/maventask.ts#L313, which we don't really need for our scenarios. Where as options actually get passed to the maven command line. 

As for the logging we might want to move those to the options instead but I couldn't find anything definitive that lead me to believe it was useful @lenala can you speak to whether or not those logging options are interesting for us?

Reply:
>Is it valid to move -Dmaven.test.skip=true from mavenOptions to options?

Yes this is valid it probably shouldn't have been in the mavenOptions to begin with.

Reply:
OK after further digging on the logging options I think we should exclude these options as they reduce the log output to only errors and only warnings for transfering (i.e. downloading). Personally for CI I like to see more logging as opposed to less in case something goes wrong we can more easily debug it. If we find there is too much noise we can revisit this decision.

Reply:
I do know we want to keep the `--batch-mode` option, since without it the logs are very noisy.  I don't know anything about the other logging options.

Reply:
`--batch-mode` itself doesn't do much about logging it is mostly to ensure you don't block for any input at the console. `-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn` controls the downloading messages, it only works in batch mode and setting it to warn eliminates a lot of noise about which packages are downloading. `-Dorg.slf4j.simpleLogger.defaultLogLevel=error` tells maven to only output errors but honestly I think having more messages on the log is useful I wish we could the more diagnostic logging to a file and keep the console logging minimal but I've not figured out a way to do that yet.